### PR TITLE
Fries for crobs

### PR DESCRIPTION
**Pull request by etcadinfinitum**
Add reaction functionality for posts.

Could use a few more changes:

- [ ] safety checking for presence + truth of all required form elements
- [x] add an icon for indicating fries have been given :fries: - fixed in 3874e2a5652356c8cb04ad989d1b3c4ad097cc78
- [x] better styling - fixed in 3874e2a5652356c8cb04ad989d1b3c4ad097cc78
- [x] align button and icon into a row instead of being stacked - fixed in 3874e2a5652356c8cb04ad989d1b3c4ad097cc78

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -120,6 +120,25 @@ def caw():
     # not logged in, yell at them to log in
     return redirect('/')
 
+@app.route('/give_fry', methods=['POST'])
+def fries_for_crobs():
+    post_id = ''
```
Comment (Chris-Johnston) [inline src/main.py]:
```suggestion
```

File: src/main.py [hunk 2]
```diff
@@ -120,6 +120,25 @@ def caw():
...
+    if 'user_id' in session and session['user_id']:
+        # logged in
+        if 'postiboi' in request.form and request.form['postiboi']:
+            post_id = request.form.get('postiboi')
+            fries = request.form.get('frybois')
+            print("post id is " + str(post_id))
```
Comment (Chris-Johnston) [inline src/main.py]:
```suggestion
```
would remove these print statements when done (same can be said for the rest of the code)

File: src/main.py [hunk 3]
```diff
@@ -120,6 +120,25 @@ def caw():
...
+            print(type(post_id))
+            print("current fry count is " + str(fries))
+            print(type(fries))
+            with app.app_context():
+                db = get_db()
+                cur = db.cursor()
+                cur.execute('''UPDATE posts SET fries = ? WHERE id = ?''', (str(int(fries) + 1), post_id))
```
Comment (Chris-Johnston) [inline src/main.py]:
```suggestion
                cur.execute('''UPDATE posts SET fries = ? WHERE id = ?''', (int(fries) + 1, post_id))
```
this should be fine?

File: src/main.py [hunk 4]
```diff
@@ -120,6 +120,25 @@ def caw():
...
+                db.commit()
+    return redirect('/')
```
Comment (Chris-Johnston) [inline src/main.py]: if we wanted to do this without html `<form>` reloading the page, we could start to add JS in the front-end

File: src/post.py [hunk 5]
```diff
@@ -7,14 +7,13 @@
 import random
 
 class Post():
-    def __init__(self, author: str = "No Author", contents: str = "CAW CAW CAW", points: int = -1):
+    def __init__(self, idnum: int = -1, author: str = "No Author", contents: str = "CAW CAW CAW", points: int = -1):
```
Comment (Chris-Johnston) [inline src/post.py]:
```suggestion
    def __init__(self, id: int = -1, author: str = "No Author", contents: str = "CAW CAW CAW", points: int = -1):
```
I just prefer this naming

File: src/post.py [hunk 6]
```diff
@@ -7,14 +7,13 @@
...
+        self.idnum = idnum
```
Comment (Chris-Johnston) [inline src/post.py]:
```suggestion
        self.id = id
```

File: src/templates/homepage.html [hunk 7]
```diff
@@ -26,6 +26,15 @@ <h3>Tell your fellow crobs how you're cawwing today.</h2>
                     <div class="author">@{{ p.author }}</div>
                     <!-- TODO CROBATARS -->
                     <div class="message">{{ p.contents }}</div>
+                    <div class="fries">
+                        <form method="POST" action="/give_fry">
+                            <input readonly type="hidden" value="{{ p.idnum }}" name="postiboi"></input>
```
Comment (Chris-Johnston) [inline src/templates/homepage.html]:
```suggestion
                            <input readonly type="hidden" value="{{ p.id}}" name="postiboi"></input>
```

File: src/templates/homepage.html [hunk 8]
```diff
@@ -26,6 +26,15 @@ <h3>Tell your fellow crobs how you're cawwing today.</h2>
...
+                            <input readonly type="hidden" value="{{ p.points }}" name="frybois"></input>
+                            <!-- to do: dynamically say fry/fries depending on quantity -->
```
Comment (Chris-Johnston) [inline src/templates/homepage.html]: good news, this can be done statically, no js required.

File: src/post.py (hunk 5 above)
Comment (etcadinfinitum) [reply src/post.py]: `id` is a reserved keyword in Python, which is why I didn't use `id` to begin with. I agree that `id` is better, but would rather not appropriate a python keyword (even though it is technically legal.

File: src/post.py (hunk 6 above)
Comment (etcadinfinitum) [reply src/post.py]: see comment for previous line

File: src/templates/homepage.html (hunk 7 above)
Comment (etcadinfinitum) [reply src/templates/homepage.html]: see comment in suggested changes for `post.py`:
> id is a reserved keyword in Python, which is why I didn't use id to begin with. I agree that id is better, but would rather not appropriate a python keyword (even though it is technically legal.

File: src/main.py (hunk 4 above)
Comment (etcadinfinitum) [reply src/main.py]: This would be good, not sure how to go about implementing it but will mull on it

File: src/templates/homepage.html (hunk 8 above)
Comment (etcadinfinitum) [reply src/templates/homepage.html]: This would be good, not sure how to go about implementing it but will mull on it

Comment (etcadinfinitum) [reply src/templates/homepage.html]: resolved in bc41583a011b6515377295713479b8e074340108
used a conditional in the HTML template to render the `<p>` element based on the value

**Comment by etcadinfinitum** [comment]
@Chris-Johnston let me know what you think about the naming conventions - I'm okay changing them if you _really_ think we should, but I'm not inclined to make a keyword into a variable just in case something breaks down the line.

Regarding the page reload - I'm still not sure how to go about implementing this, but would be okay making this into its own feature branch & PR due to addition of JS. 

In the interim, check out the majesty that is the new reaction section:
new_crobber_fry_stylings

File: src/templates/homepage.html (hunk 7 above)
Comment (Chris-Johnston) [reply src/templates/homepage.html]: huh. wasn't aware that was a reserved keyword. let's not change it then.

File: src/main.py (hunk 4 above)
Comment (Chris-Johnston) [reply src/main.py]: XMLHttpRequest with a callback, although that could be the start of a very large rabbit hole

**Pull request merged**